test/main/udf_test.cpp

```diff
@@ -19,6 +29,22 @@ TEST_F(ApiTest, UnaryUDFInt64) {
     sortAndCheckTestResults(actualResult, expectedResult);
 }
 
+TEST_F(ApiTest, TestDropUDF) {
+    conn->createScalarFunction("add5", &add5);
+    conn->removeUDFFunction("add5");
+    conn->createScalarFunction("add5", &add5);
+}
+
+TEST_F(ApiTest, TestExceptUDF) {
+    conn->createScalarFunction("test", &throwRuntimeException);
+    try {
+        conn->query("RETURN test()");
+    } catch (RuntimeException) {}
```

Comment:
i think the exception will never be catched. We always store the exception message inside the queryresult rather than throw to the user. You have to check the queryResult instead of using a catch statement